python/pyspark/pandas/tests/test_ops_on_diff_frames.py

```diff
@@ -1842,6 +1842,38 @@ def _test_cov(self, pser1, pser2):
         pscov = psser1.cov(psser2, min_periods=3)
         self.assert_eq(pcov, pscov, almost=True)
 
+    def test_corrwith(self):
+        df1 = pd.DataFrame({"A": [1, np.nan, 7, 8], "X": [5, 8, np.nan, 3], "C": [10, 4, 9, 3]})
+        df2 = df1[["A", "C"]]
+        self._test_corrwith(df1, df2)
+        self._test_corrwith(df1, df2.A)
+
+        df3 = pd.DataFrame({"A": [5, 3, 6, 4], "B": [11, 2, 4, 3], "C": [4, 3, 8, np.nan]})
+        self._test_corrwith(df1, df3)
+        self._test_corrwith(df1, df3.B)
+
+        df_bool = pd.DataFrame({"A": [True, True, False, False], "B": [True, False, False, True]})
+        ser_bool = pd.Series([True, True, False, True])
+        self._test_corrwith(df_bool, ser_bool)
+
+        self._test_corrwith(self.pdf1, self.pdf1)
+        self._test_corrwith(self.pdf1, self.pdf2)
+        self._test_corrwith(self.pdf2, self.pdf3)
+        self._test_corrwith(self.pdf3, self.pdf4)
+
+        self._test_corrwith(self.pdf1, self.pdf1.a)
+        self._test_corrwith(self.pdf1, self.pdf2.b)
+        self._test_corrwith(self.pdf2, self.pdf3.c)
+        self._test_corrwith(self.pdf3, self.pdf4.f)
+
+    def _test_corrwith(self, pdf, pobj):
+        psdf = ps.from_pandas(pdf)
+        psobj = ps.from_pandas(pobj)
```

Comment:
Can we use one way for `_test_corrwith` ??

In `test_dataframe.py`, `_test_corrwith` takes pandas-on-Spark DataFrame and convert it to pandas, but here it takes pandas DataFrame and convert it to pandas-on-Spark DataFrame, so I think maybe it's a bit confused.

Reply:
They were different since:
1, in `test_ops_on_diff_frames.py`, I want to reuset existing pandas' dafaframes `self.pdf?`
2, in `test_dataframe.py`, I need to make sure both inputs share the same anchor.

Reply:
Got it. Let's keep it as is

Reply:
my bad, just found that there already exists pandas-on-Spark DataFrame  `self.psdf?` in `test_ops_on_diff_frames.py`, will make them the same